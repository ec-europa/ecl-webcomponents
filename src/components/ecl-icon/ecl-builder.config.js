const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
