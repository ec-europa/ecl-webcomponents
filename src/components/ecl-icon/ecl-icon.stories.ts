import iconPathEc from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
import iconPathEu from '@ecl/resources-eu-icons/dist/sprites/icons.svg';
import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

export default {
  title: 'Components/icon',
};

const Template = (args) => `<ecl-icon 
                              icon="${args.icon}"
                              color="${args.color}"
                              size="${args.size}"
                              transform="${args.transform}"
                              path="${args.theme === 'ec' ? iconPathEc : iconPathEu}"
                            ></ecl-icon>`;

export const Icon = Template.bind({});
Icon.storyName = 'default';
Icon.args = {
  theme: 'ec',
  icon: 'close',
  transform: '',
  color: 'default',
  size: '2xl',
};
Icon.argTypes = {
  theme: {
    type: { name: 'select'},
    options: ['ec', 'eu'],
  },
  icon: {
    type: { name: 'select' },
    options: iconsAllEc,
    description: "Add an icon",
  },
  size: {
    type: { name: 'select' },
    options: ['2xs','xs', 's', 'm', 'l', 'xl', '2xl'],
    description: "icon size",
  },
  color: {
    type: { name: 'select' },
    options: ['default', 'primary', 'inverted'],
    description: "icon color",
  },
  transform: {
    type: { name: 'select' },
    description: 'Button icon transform',
    options: [
      'rotate-90',
      'rotate-180',
      'rotate-270',
      'flip-horizontal',
      'flip-vertical',
    ],
  },
};