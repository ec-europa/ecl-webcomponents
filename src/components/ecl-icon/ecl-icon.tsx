import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}