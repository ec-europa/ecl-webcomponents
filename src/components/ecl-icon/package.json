{
  "name": "@ecl/web-components-icon",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.12.0",
  "description": "ECL Icon",
  "main": "ecl-icon.ts",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:copy": "cross-env ecl-builder copy",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/resources-ec-icons": "3.13.0",
    "@ecl/resources-eu-icons": "3.13.0",
    "@ecl/specs-component-icon": "3.13.0",
    "@ecl/vanilla-component-icon": "3.13.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
