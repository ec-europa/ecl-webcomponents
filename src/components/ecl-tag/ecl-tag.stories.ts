const getArgs = () => {
  return {
    variant: 'display',
    url: '',
    external: '',
    ariaLabel: 'aria label',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['link', 'removable', 'display'],
      description: "Tag variant"
    },
    external: {
      name: 'external',
      type: { name: 'boolean' },
      description: "When the link is external"
    },
    ariaLabel: {
      name: 'aria label',
      type: { name: 'string' },
      description: 'To improve accessibility',
    }
  };
};

export default {
  title: 'Components/tag',
};

const Template = (args) => 
`<ecl-tag
  variant="${args.variant}"
  theme="${args.theme}"
  url="${args.url}"
  aria-label="${args.ariaLabel}"
  external="${args.external}"
>
  An ECL tag
</ecl-tag>`;

export const Tag = Template.bind({});
Tag.storyName = 'default';
Tag.args = getArgs();
Tag.argTypes = getArgTypes();