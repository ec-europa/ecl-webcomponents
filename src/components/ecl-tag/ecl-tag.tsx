import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-tag',
  styleUrls: {
    ec: './build/styles/ecl-tag-ec.css',
    eu: './build/styles/ecl-tag-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclTag {
  @Prop() styleClass: string;
  @Prop() external: boolean = false;
  @Prop() variant: string = 'display';

  getClass(): string {
    return [
      `ecl-tag`,
      `ecl-tag--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  componentWillLoad() {
    if (this.variant === 'link' || this.variant === 'removable') {
      const style = document.createElement('style');
      if (this.variant === 'link') {    
        style.innerHTML = `.ecl-tag a { color: #fff; }
                            .ecl-tag__icon--external {
                              margin-inline-start: 0.5rem !important;
                              position: relative;
                              top: 1px;
                            }`;
      } else {
        style.innerHTML = `.ecl-tag .ecl-icon { fill: #fff; }
                          .ecl-tag__icon-close {
                            fill: map.get(theme.$color, 'white');
                            left: 0;
                            opacity: 1;
                            position: absolute;
                            top: 0;
                          }
                          .ecl-tag__icon-close-filled {
                            fill: map.get(theme.$color, 'white');
                            left: 0;
                            opacity: 0;
                            position: absolute;
                            top: 0;
                          };`
      }

      document.body.appendChild(style);
    }
  }

  getTag(variant) {
    switch (variant) {
      case 'link':
        return 'a'

      case 'display':
        return 'span';

      case 'removable':
        return 'button';

       default:
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class="ecl-tag__icon ecl-tag__icon--external"
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  getRemovable() {
    return (
      <span
        class="ecl-tag__icon"
      >
        <ecl-icon 
          icon="close"
          style-class="ecl-tag__icon-close"
        >
        </ecl-icon>
        <ecl-icon
          icon="close-filled"
          style-class="ecl-tag__icon-close-filled"
        ></ecl-icon>
      </span>
    )
  }

  render() {
    const Element = this.getTag(this.variant);
    return (
      <Element 
        class={this.getClass()}
      >
        <slot></slot>
        { this.variant === 'link' && this.external ? <this.getExternal /> : '' }
        { this.variant === 'removable' ? <this.getRemovable /> : '' }
      </Element>
    )
  }
}