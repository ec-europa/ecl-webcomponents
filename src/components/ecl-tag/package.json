{
  "name": "@ecl/web-components-tag",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.9.0",
  "description": "ECL Tag",
  "main": "ecl-tag.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.9.1",
    "@ecl/resources-ec-icons": "3.9.1",
    "@ecl/resources-eu-icons": "3.9.1",
    "@ecl/specs-component-icon": "3.9.1",
    "@ecl/vanilla-component-icon": "3.9.1",
    "@ecl/vanilla-component-tag": "3.9.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
