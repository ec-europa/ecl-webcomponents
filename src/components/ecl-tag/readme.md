# ecl-tag



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default     |
| ------------ | ------------- | ----------- | --------- | ----------- |
| `external`   | `external`    |             | `boolean` | `false`     |
| `styleClass` | `style-class` |             | `string`  | `undefined` |
| `theme`      | `theme`       |             | `string`  | `'ec'`      |
| `variant`    | `variant`     |             | `string`  | `'display'` |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-tag --> ecl-icon
  style ecl-tag fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
