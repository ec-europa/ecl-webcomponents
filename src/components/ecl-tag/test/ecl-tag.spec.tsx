import { newSpecPage } from '@stencil/core/testing';
import { EclTag } from '../ecl-tag';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-tag', () => {
  it('renders correctly in the display variant in EC', async () => {
    const page = await newSpecPage({
      components: [EclTag],
      html: `<ecl-tag
              type="display"
              data-ecl-button-icon
              theme="ec"
            >
              An ECL tag
            </ecl-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('renders correctly in the display variant in EU', async () => {
    const page = await newSpecPage({
      components: [EclTag],
      html: `<ecl-tag
              type="display"
              data-ecl-button-icon
              theme="ec"
            >
              An ECL tag
            </ecl-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('renders correctly in the link variant in EC', async () => {
    const page = await newSpecPage({
      components: [EclTag],
      html: `<ecl-tag
              type="link"
              url="/example"
              theme="ec"
            >
              An ECL tag
            </ecl-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('renders correctly in the link variant in EU', async () => {
    const page = await newSpecPage({
      components: [EclTag],
      html: `<ecl-tag
              type="link"
              url="/example"
              theme="eu"
            >
              An ECL tag
            </ecl-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('renders correctly in the removable variant in EC', async () => {
    const page = await newSpecPage({
      components: [EclTag, EclIcon],
      html: `<ecl-tag
              type="removable"
              theme="ec"
            >
              An ECL tag
            </ecl-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('renders correctly in the removable variant in EU', async () => {
    const page = await newSpecPage({
      components: [EclTag, EclIcon],
      html: `<ecl-tag
              type="removable"
              theme="eu"
            >
              An ECL tag
            </ecl-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
