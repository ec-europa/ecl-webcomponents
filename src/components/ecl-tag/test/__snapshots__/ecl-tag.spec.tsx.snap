// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-tag renders correctly in the display variant in EC 1`] = `
<ecl-tag data-ecl-button-icon theme="ec" type="display">
  <!---->
  <span class="ecl-tag ecl-tag--display">
    An ECL tag
  </span>
</ecl-tag>
`;

exports[`ecl-tag renders correctly in the display variant in EU 1`] = `
<ecl-tag data-ecl-button-icon theme="ec" type="display">
  <!---->
  <span class="ecl-tag ecl-tag--display">
    An ECL tag
  </span>
</ecl-tag>
`;

exports[`ecl-tag renders correctly in the link variant in EC 1`] = `
<ecl-tag theme="ec" type="link" url="/example">
  <!---->
  <span class="ecl-tag ecl-tag--display">
    An ECL tag
  </span>
</ecl-tag>
`;

exports[`ecl-tag renders correctly in the link variant in EU 1`] = `
<ecl-tag theme="eu" type="link" url="/example">
  <!---->
  <span class="ecl-tag ecl-tag--display">
    An ECL tag
  </span>
</ecl-tag>
`;

exports[`ecl-tag renders correctly in the removable variant in EC 1`] = `
<ecl-tag theme="ec" type="removable">
  <!---->
  <span class="ecl-tag ecl-tag--display">
    An ECL tag
  </span>
</ecl-tag>
`;

exports[`ecl-tag renders correctly in the removable variant in EU 1`] = `
<ecl-tag theme="eu" type="removable">
  <!---->
  <span class="ecl-tag ecl-tag--display">
    An ECL tag
  </span>
</ecl-tag>
`;
